akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/javadsl/EventSourcedEntity.scala

```diff
@@ -43,3 +48,40 @@ abstract class EventSourcedEntity[Command, Event, State] private (
   }
 
 }
+
+/**
+ * Any [[Behavior]] can be used as a sharded entity actor, but the combination of sharding and persistent
+ * actors is very common and therefore this `PersistentEntity` class is provided as convenience.
+ *
+ * A [[EventSourcedEntityWithEnforcedReplies]] enforces that replies to commands are not forgotten.
+ * There will be compilation errors if the returned effect isn't a [[ReplyEffect]], which can be
+ * created with `Effects().reply`, `Effects().noReply`, [[Effect.thenReply]], or [[Effect.thenNoReply]].
+ *
+ * It is a [[EventSourcedBehavior]] and is implemented in the same way. It selects the `persistenceId`
+ * automatically from the [[EntityTypeKey]] and `entityId` constructor parameters by using
+ * [[EntityTypeKey.persistenceIdFrom]].
+ */
+abstract class EventSourcedEntityWithEnforcedReplies[Command, Event, State] private (
+    val entityTypeKey: EntityTypeKey[Command],
+    val entityId: String,
+    persistenceId: PersistenceId,
+    onPersistFailure: Optional[BackoffSupervisorStrategy])
+    extends EventSourcedBehaviorWithEnforcedReplies[Command, Event, State](persistenceId, onPersistFailure) {
+
+  def this(entityTypeKey: EntityTypeKey[Command], entityId: String) = {
+    this(
+      entityTypeKey,
+      entityId,
+      persistenceId = entityTypeKey.persistenceIdFrom(entityId),
+      Optional.empty[BackoffSupervisorStrategy])
+  }
+
+  def this(entityTypeKey: EntityTypeKey[Command], entityId: String, onPersistFailure: BackoffSupervisorStrategy) = {
+    this(
+      entityTypeKey,
+      entityId,
+      persistenceId = entityTypeKey.persistenceIdFrom(entityId),
+      Optional.ofNullable(onPersistFailure))
+  }
+
+}
```

Comment:
I can add those on this same PR and some test to prove that things compile at least. Maybe not necessary to write a full functional test for all those cases because we are only doing type massage.

Reply:
Yes, compilation tests would be enough

Reply:
I added a compilation only test for the enforced replies case.

And `of` variants for it as well.